pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -112,3 +129,40 @@ func (f *framework) RunReservePlugins(
 func (f *framework) NodeInfoSnapshot() *cache.NodeInfoSnapshot {
 	return f.nodeInfoSnapshot
 }
+
+func pluginConfigToMap(args []config.PluginConfig) map[string]*runtime.Unknown {
+	pc := make(map[string]*runtime.Unknown, 0)
+	for _, p := range args {
+		pc[p.Name] = &p.Args
+	}
+	return pc
+}
+
+func pluginNeeded(plugins config.Plugins) map[string]bool {
+	pg := map[string][]config.Plugin{
+		"QueueSort":      plugins.QueueSort,
+		"PreFilter":      plugins.PostFilter,
+		"Filter":         plugins.Filter,
+		"PostFilter":     plugins.PostFilter,
+		"Score":          plugins.Score,
+		"NormalizeScore": plugins.NormalizeScore,
+		"Reserve":        plugins.Reserve,
+		"Permit":         plugins.Permit,
+		"PreBind":        plugins.PreBind,
+		"Bind":           plugins.PostBind,
+		"PostBind":       plugins.PostBind,
+		"UnReserve":      plugins.UnReserve,
+	}
+
+	pgMap := make(map[string]bool, 0)
+	fn := func(pg []config.Plugin) {
```

Comment:
Why do we need this inline function? Could you please place the body of this function directly in the following for loop?